src/CNZTU.CPP

```diff
@@ -1569,14 +1573,35 @@ RC ZNR::ztuEndSubhr()	// end-subhour (after iterations) hvac checks/computations
 //  - unMetMaxTD = largest excursion, F
 	if (zn_tzsp > 0.f)		// if setpoint active
 	{
+		int iHC = -1;	// -1=no HVAC, 0=heating, 1=cooling
+		if (zn_HasTerminal())
+		{
+			const ZHX* zhx = ZhxB.GetAtSafe(mdSeq[zn_md]);
+			if (zhx)
+			{
+#if defined( _DEBUG)
+				if (zhx->sp != zn_tzsp)
+					printf("\nSP mismatch");
+#endif
+				if (zhx->zhxTy & StH)
+					iHC = 0;
+				else if (zhx->zhxTy & StC)
+					iHC = 1;
+			}
+		}
+		else if (zn_hcMode == RSYS::rsmHEAT
+			 || zn_hcMode == RSYS::rsmCOOL)
+		{
+			iHC = zn_hcMode - 1;
+			static_assert(RSYS::rsmHEAT == 1);
+			static_assert(RSYS::rsmCOOL == 2);
+		}
+
 		float tzErr = tz - zn_tzsp;		// temp excursion above/below setpoint
 
-		if ((zn_hcMode == RSYS::rsmHEAT && tzErr < -0.001f)
-		 || (zn_hcMode == RSYS::rsmCOOL && tzErr > 0.001f))
+		if ((iHC == 0 && tzErr < -0.001f)
```

Comment:
Should we use enums for this?

Reply:
Maybe.  iHC is idx into ZNRES arrays.  It would be good to share any enum definitions such that they are accessible in user expressions.  I'll think about how to do that.